kite-data/kite-data-core/src/main/java/org/kitesdk/data/spi/filesystem/FileSystemWriter.java

```diff
@@ -148,64 +169,74 @@ public final void write(E entity) {
     }
   }
 
+  /**
+   * @deprecated will be removed in 1.0.0; use Flushable#flush instead
+   */
   @Override
+  @Deprecated
   public void flush() {
-    Preconditions.checkState(state.equals(ReaderWriterState.OPEN),
-        "Attempt to flush a writer in state:%s", state);
-    try {
-      appender.flush();
-    } catch (IOException e) {
-      this.state = ReaderWriterState.ERROR;
-      throw new DatasetWriterException("Failed to flush appender " + appender);
-    }
```

Comment:
Why has the implementation been removed here? Shouldn't it stay until after 0.18?